File: Sleep Sort/SleepSort.c
#include <stdio.h>
#include <windows.h>

void sleepSort(int arr[], int n) {

    for(int i = 0; i < n; i++) {

        Sleep(arr[i] * 100);

        printf("%d ", arr[i]);
    }
}

int main() {
    int arr[] = {4, 2, 7, 1, 3};
    int n = 5;

    sleepSort(arr, n);

    return 0;
}

File: Sleep Sort/SleepSort.py
import threading
import time

def sleep_sort(num):
    time.sleep(num * 0.1)
    print(num, end=" ")

arr = [4, 2, 7, 1, 3]

threads = []

for num in arr:
    t = threading.Thread(target=sleep_sort, args=(num,))
    threads.append(t)
    t.start()

for t in threads:
    t.join()
